crates/snp_validation/src/lib.rs

```diff
@@ -287,96 +308,129 @@ impl Snp {
         }
 
-        for value in report.body.host_data {
-            if value != 0 {
-                return Err(anyhow!("snp report host_data field not set by Enarx"));
-            }
+        ensure!(
+            report.body.vmpl == 0,
+            "snp report vmpl field not set by Enarx"
+        );
+
+        // Check field set by Enarx
+        for value in report.body.report_id_ma {
+            ensure!(
+                value == 255,
+                "snp report report_id_ma field not the value set by Enarx"
+            );
         }
 
-        for value in report.body.id_key_digest {
-            if value != 0 {
-                return Err(anyhow!("snp report id_key_digest field not set by Enarx"));
+        if let Some(config) = config {
+            ensure!(
+                config.minimum_abi.matches(&report.body.policy.into()),
+                "snp minimum abi not met"
+            );
+
+            if !config.enarx_measurements.signer().is_empty() {
+                let signed = config
+                    .enarx_measurements
+                    .signer()
+                    .iter()
+                    .any(|s| report.body.author_key_digest == s.as_ref());
+                ensure!(signed, "snp untrusted enarx author_key");
             }
-        }
 
-        if report.body.vmpl != 0 {
-            return Err(anyhow!("snp report vmpl field not set by Enarx"));
-        }
+            if !config.id_key_digest.is_empty() {
+                let signed = config
+                    .id_key_digest
+                    .iter()
+                    .any(|id| report.body.id_key_digest == id.as_ref());
+                ensure!(signed, "snp untrusted enarx id_key");
+            }
 
-        if report.body.guest_svn != 0 {
-            return Err(anyhow!("snp report guest_svn field not set by Enarx"));
-        }
+            if !config.id_key_digest_blacklist.is_empty() {
+                for signer in &config.id_key_digest_blacklist {
+                    if report.body.id_key_digest == signer.as_ref() {
+                        bail!("snp untrusted enarx id_key");
+                    }
+                }
+            }
 
-        // Check field set by Enarx
-        for value in report.body.report_id_ma {
-            if value != 255 {
-                return Err(anyhow!(
-                    "snp report report_id_ma field not the value set by Enarx"
-                ));
+            if !config.enarx_measurements.hash().is_empty() {
+                let allowed = config
+                    .enarx_measurements
+                    .hash()
+                    .iter()
+                    .any(|h| report.body.measurement == h.as_ref());
+                ensure!(allowed, "snp untrusted enarx hash");
+            }
+
+            if let Some(flags) = config.platform_info_flags {
+                ensure!(
+                    flags == report.body.plat_info.flag,
+                    "snp unexpected memory mode"
```

Comment:
Yes, it's about memory only currently, but that can change. Also it is clear now where to look.
```suggestion
                    "snp unexpected platform info flags"
```